server/blob/blob-s3/src/main/java/org/apache/james/blob/objectstorage/aws/S3BlobStoreDAO.java

```diff
@@ -111,6 +114,11 @@
             .maxPendingAcquireUnbounded()
             .sizeUnbounded()
             .fifo();
+
+        bucketNameResolver = BucketNameResolver.builder()
```

Comment:
Maybe we can inject the all bucketname resolver stuff?

Reply:
I thought here that: 
* we only use bucket name resolver in S3BlobStoreDAO
* the bucket name resolver is being built with namespace + bucket prefix defined in conf
* I think what we needed until now was aws s3 auth conf + region conf... but if we had the bucket prefix and namespace for the resolver, then S3BlobStore needs all params from S3BlobStoreConfiguration. 

So I thought injecting directly the all conf and doing the stuff I need in the constructor of S3BlobStore was more simple. Allowed to get rid of the region and awsS3Auth Guice bidings in S3BlobStoreModule (as we don't seem to use them anywhere else). The integration tests are not breaking locally

Reply:
Let's keep this dumb simple then ;-)